src/SMARTPL.g

```diff
@@ -82,6 +82,9 @@ STRTAG		:	'artist'
 			|	'path'
 			|	'type'
 			|	'grouping'
+			|	'artist_id'
+			|	'album_id'
```

Comment:
Why are "artist_id" and "album_id" added to the query language?

Reply:
Makes sense - let me re-org the PR into 4 commits + last for generation of `htdocs` - it might take a few days.  

Regarding your comments:
* query language
it is useful to search based on these ids if they are available (I've done this to query the backend through `curl` and then `jq` to avoid using sqlite on the db).  Is this not an expected use case of the query language?
An example:  getting more info (like tracks) for an album called "The Greatest Hits" for a specific artist.  via curl, get the artist album info and use the `album_id` to pull the tracks/info for that album instead of doing another query for artist and album = "The Greatest Hits".
But you are right .. the `artist_id` and `album_id` are not used directly in any of the code submitted.
* infinite scroll
your observations in your repo re: slowness on listing large number of albums was indeed the rendering at the client side - the infinite scroll is here to provide the _loading spinner_;  I load a small number of elements, render, show the spinner (instead of what could look like a hung session) and then immediately get the rest of data to render.  This was only useful for my slow slow cell phone which I've got rid of and we can remove this if this is too much.
On my newer cell phone, the render is quicker and I never see the spinner.
The index tab navigation has no issues with going fwd and then back in history (on firefox and ios safari) once all the data is loaded and rendered.  The browsers even maintain the position of the page (say I've use the indx to scroll to an anchor near bottom of hte page)